### PR TITLE
Fix/Notes-48420: Make sure to display draft notes in the tree view drawer and fix process of saving drafts.

### PR DESCRIPTION
**Pull request by mdiha**

### CODE REVIEW

**Review by SaraBoutej — APPROVED** [review]

File: notes-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java [hunk 1]
```diff
@@ -287,6 +288,19 @@ public Page getParentPageOf(Page page) throws WikiException {
         childrenPages.add(convertPageEntityToPage(childPageEntity));
       }
     }
+    boolean isRootPage = pageEntity.getParentPage() == null;
+    List<DraftPageEntity> draftPageEntities;
+    if (isRootPage) {
+      draftPageEntities = draftPageDAO.findRootDraftPagesByUser(ConversationState.getCurrent().getIdentity().getUserId());
```
Comment (ahamdi) [inline notes-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java:294]: ConversationState should not be used in Storage layer

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java [hunk 2]
```diff
@@ -823,6 +823,9 @@ public boolean hasPermissionOnWiki(Wiki wiki, PermissionType permissionType, Ide
 
   @Override
   public boolean hasPermissionOnPage(Page page, PermissionType permissionType, Identity user) throws WikiException {
+    if (page.isDraftPage()) {
+      return page.getAuthor().equals(ConversationState.getCurrent().getIdentity().getUserId());
```
Comment (ahamdi) [inline notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:827]: why using ConversationState, whil eyou have the user Identity ?

File: notes-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java (hunk 1 above)
Comment (plamarque) [reply notes-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java:294]: It seems important architecturally. Do we have an item for this in the dev guidelines ?
CC @boubaker

Comment (mdiha) [reply notes-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java:294]: I'am fixing this in the PR to be merged to develop, sorry i didn't see the comment. cc @ahamdi

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java (hunk 2 above)
Comment (mdiha) [reply notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:827]: Yes you are right.

**Pull request merged**